active/CornerRadius/CornerRadiusToThickness2.md

```diff
@@ -0,0 +1,149 @@
+
+# Background
+
+_This spec updates the existing [CornerRadiusToThicknessConverter](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.Primitives.CornerRadiusToThicknessConverter) class with new filters._
+
+WinUI has a [CornerRadius](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.CornerRadius) struct that's used to define the rounding the 4 corners of a box, and the [Thickness](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Thickness) struct that's used for defining the padding or margin around the 4 sides of a box.
+
+CornerRadius defines the radius length of the circle segment that's used to draw a rounded corner. For example the following uses 4 different radii for the 4 corners (the corners are listed in clockwise order starting from the top left corner):
+
+```xml
+<Border CornerRadius="0,20,40,60" BorderBrush="Red" BorderThickness="2" Height="100" Width="100" />
+```
+
+![Corner radius example](./CornerRadius.png)
+
+Thickness defines 4 sides. For example the following uses 4 different values for the margins around a blue rectangle in a red box (the sides are listed in clockwise order starting from the left):
+
+```xml
+<Border BorderBrush="Red" BorderThickness="2" Height="100" Width="100" >
+    <Rectangle Fill="Blue" Margin="0,20,40,60" />
+</Border>
+```
+
+![Thickness example](./Thickness.png)
+
+For consistency it's useful to be able to build a Thickness out of components of a CornerRadius. For this reason the [CornerRadiusFilterConverter](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.Primitives.CornerRadiusFilterConverter) exists, which can be used in Xaml markup. That converter has a [CornerRadiusFilterKind](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.Primitives.CornerRadiusFilterKind) that lets you define which parts to pull out, what's being added in this spec is update to that enum to allow more parts to be pulled out.
+
+**Existing values:**
+* FilterTopAndBottomFromLeft,
+* FilterTopAndBottomFromRight,
+* FilterLeftAndRightFromTop,
+* FilterLeftAndRightFromBottom
+
+**New values:**
+* FilterTopFromTopLeft,
+* FilterTopFromTopRight,
+* FilterRightFromTopRight,
+* FilterRightFromBottomRight,
+* FilterBottomFromBottomRight,
+* FilterBottomFromBottomLeft,
+* FilterLeftFromBottomLeft,
+* FilterLeftFromTopLeft,
+
+The CornerRadiusFilterConverter is also updated with a new property -- Multiplier -- which is applied during the conversion. For example, setting this to -1 negates the value.

```

Comment:
Would `Scale` be a more natural name than `Multiplier`?

Reply:
For strictly positive values, I would say yes, but I am not sure if you refer to a scale of "-1".

Reply:
Negative scale factors are a thing.

Reply:
That's true, though I mostly have encountered them in a very mathematical scenario. Given that negative values are not that common as use case, `Scale` might be the better wording.

Reply:
Discussion: We have two "multipliers" today (e.g. [PointerDeviceUsage.PhysicalMultiplier](https://docs.microsoft.com/uwp/api/Windows.Devices.Input.PointerDeviceUsage.PhysicalMultiplier)). Note that both are floats rather than this (Thickness). Scales are mostly scalars, vector2, or vector3s.

"Scale" might be confusing when negative, because a scale transform does a flip, which this does not.

Recommend: let's stick with multiplier